content/doc/book/managing/nodes.adoc

```diff
@@ -14,6 +14,119 @@ endif::[]
+ port = "SMTP_PORT"
+ 
+ Properties mprops = new Properties();
+ mprops.setProperty("mail.transport.protocol","smtp");
+ mprops.setProperty("mail.host",host);
+ mprops.setProperty("mail.smtp.port",port);
+ 
+ Session lSession = Session.getDefaultInstance(mprops,null);
+ MimeMessage msg = new MimeMessage(lSession);
+ 
+ 
+ //tokenize out the recipients in case they came in as a list
+ StringTokenizer tok = new StringTokenizer(toAddress,";");
+ ArrayList emailTos = new ArrayList();
+ while(tok.hasMoreElements()){
+ emailTos.add(new InternetAddress(tok.nextElement().toString()));
+ }
+ InternetAddress[] to = new InternetAddress[emailTos.size()];
+ to = (InternetAddress[]) emailTos.toArray(to);
+ msg.setRecipients(MimeMessage.RecipientType.TO,to);
+ InternetAddress fromAddr = new InternetAddress(fromAddress);
+ msg.setFrom(fromAddr);
+ msg.setFrom(new InternetAddress(fromAddress));
+ msg.setSubject(subject);
+ msg.setText(message)
+ 
+ Transport transporter = lSession.getTransport("smtp");
+ transporter.connect();
+ transporter.send(msg);
+}
+ 
+ 
+def getEnviron(computer) {
+   def env
+   def thread = Thread.start("Getting env from ${computer.name}", { env = computer.environment })
+   thread.join(2000)
+   if (thread.isAlive()) thread.interrupt()
+   env
+}
+ 
+def slaveAccessible(computer) {
+    getEnviron(computer)?.get('PATH') != null
+}
+ 
+ 
+def numberOfflineNodes = 0
+def numberNodes = 0
+for (slave in jenkins.slaves) {
+   def computer = slave.computer
+   numberNodes ++
+   println ""
+   println "Checking computer ${computer.name}:"
+   def isOK = (slaveAccessible(computer) && !computer.offline)
+   if (isOK) {
+     println "\t\tOK, got PATH back from slave ${computer.name}."
+     println('\tcomputer.isOffline: ' + slave.getComputer().isOffline()); 
+     println('\tcomputer.isTemporarilyOffline: ' + slave.getComputer().isTemporarilyOffline());
+     println('\tcomputer.getOfflineCause: ' + slave.getComputer().getOfflineCause());
+     println('\tcomputer.offline: ' + computer.offline); 
+      
+      
+   } else {
+     numberOfflineNodes ++
+     println "  ERROR: can't get PATH from slave ${computer.name}."
+     println('\tcomputer.isOffline: ' + slave.getComputer().isOffline()); 
+     println('\tcomputer.isTemporarilyOffline: ' + slave.getComputer().isTemporarilyOffline());
+     println('\tcomputer.getOfflineCause: ' + slave.getComputer().getOfflineCause());
+     println('\tcomputer.offline: ' + computer.offline); 
+     sendMail(computer.name, slave.getComputer().getOfflineCause().toString())
+     if (slave.getComputer().isTemporarilyOffline()) {
+      if (!slave.getComputer().getOfflineCause().toString().contains("Disconnected by")) {
+         computer.setTemporarilyOffline(false, slave.getComputer().getOfflineCause())        
+      }
+     } else {
+         computer.connect(true)  
+     }
+   }
+ }
+println ("Number of Offline Nodes: " + numberOfflineNodes)
+println ("Number of Nodes: " + numberNodes)  
```

Comment:
Remove trailing whitespace from this line (and from the other lines in the file.
```suggestion
println ("Number of Nodes: " + numberNodes)
```